### PR TITLE
Add support for "performed_via_github_app" and "requested_team" in IssueEvent

### PR DESCRIPTION
**Pull request by Shyclyde**
# Issue Source
#2945 mentions the reasoning for this PR.

# Why

`"performed_via_github_app"` and `"requested_team"` are valuable results in the issue events response (when available) to determine more about the event. It also keeps the struct up to date with the API response.

# What

- Adds these values to the IssueEvent struct.
- Adds test cases for added values.
- Restructures relevant test cases slightly to match the order of the given/want for easier reference.

# Other notes

Ran all available test scripts, updated accordingly. Tests passing.

### CODE REVIEW

**Review by gmlewis — APPROVED** [review]
Thank you, @Shyclyde !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by WillAbides — APPROVED** [review]

**Comment by gmlewis** [comment]
Thank you, @WillAbides !
Merging.

**Pull request merged**
